project-chip/connectedhomeip
# Allow configure nlfaultinjection via fault-cluster in yaml

#### Problem
What is being fixed?  Examples:
* Nest Labs Fault Injection (nlfaultinjection) is designed to provide a simple, portable fault injection framework, this framework has been integrated in Matter SDK from the beginning, but not get activated anywhere. We need a way to configure these faults from chip-tool so that we can use this mechanism in yaml test.

* Fixes #21941

#### Change overview
Allow configure nlfaultinjection via fault-cluster in yaml

#### Testing
How was this tested? (at least one bullet point required)
* Build all-clusters-app with nlfaultinjection enabled
`scripts/examples/gn_build_example.sh examples/all-clusters-app/linux out/debug/standalone chip_config_network_layer_ble=false chip_with_nlfaultinjection=true`

* Configure a system fault from chip_tool
`./chip-tool faultinjection fail-at-fault 1 1 0 1 true 12344321 0`

* Confirm the fault is activated from all-clusters-app log
```
[1660712608.337635][30773:30773] CHIP:EM: Received message of type 0x8 with protocolId (0, 1) and MessageCounter:143814682 on exchange 53287r
[1660712608.337649][30773:30773] CHIP:EM: Handling via exchange: 53287r, Delegate: 0x55deb7ffabc8
[1660712608.337670][30773:30773] CHIP:DMG: InvokeRequestMessage =
[1660712608.337676][30773:30773] CHIP:DMG: {
[1660712608.337682][30773:30773] CHIP:DMG: 	suppressResponse = false, 
[1660712608.337689][30773:30773] CHIP:DMG: 	timedRequest = false, 
[1660712608.337694][30773:30773] CHIP:DMG: 	InvokeRequests =
[1660712608.337703][30773:30773] CHIP:DMG: 	[
[1660712608.337707][30773:30773] CHIP:DMG: 		CommandDataIB =
[1660712608.337713][30773:30773] CHIP:DMG: 		{
[1660712608.337718][30773:30773] CHIP:DMG: 			CommandPathIB =
[1660712608.337724][30773:30773] CHIP:DMG: 			{
[1660712608.337730][30773:30773] CHIP:DMG: 				EndpointId = 0x0,
[1660712608.337735][30773:30773] CHIP:DMG: 				ClusterId = 0xfff1fc06,
[1660712608.337741][30773:30773] CHIP:DMG: 				CommandId = 0x0,
[1660712608.337746][30773:30773] CHIP:DMG: 			},
[1660712608.337752][30773:30773] CHIP:DMG: 			
[1660712608.337758][30773:30773] CHIP:DMG: 			CommandFields = 
[1660712608.337763][30773:30773] CHIP:DMG: 			{
[1660712608.337769][30773:30773] CHIP:DMG: 				0x0 = 1, 
[1660712608.337775][30773:30773] CHIP:DMG: 				0x1 = 1, 
[1660712608.337781][30773:30773] CHIP:DMG: 				0x2 = 0, 
[1660712608.337787][30773:30773] CHIP:DMG: 				0x3 = 1, 
[1660712608.337793][30773:30773] CHIP:DMG: 				0x4 = true, 
[1660712608.337798][30773:30773] CHIP:DMG: 			},
[1660712608.337803][30773:30773] CHIP:DMG: 		},
[1660712608.337811][30773:30773] CHIP:DMG: 		
[1660712608.337816][30773:30773] CHIP:DMG: 	],
[1660712608.337825][30773:30773] CHIP:DMG: 	
[1660712608.337830][30773:30773] CHIP:DMG: 	InteractionModelRevision = 1
[1660712608.337836][30773:30773] CHIP:DMG: },
[1660712608.337860][30773:30773] CHIP:DMG: AccessControl: checking f=1 a=c s=0x000000000001B669 t= c=0xFFF1_FC06 e=0 p=m
[1660712608.337869][30773:30773] CHIP:DMG: AccessControl: allowed
[1660712608.337878][30773:30773] CHIP:DMG: Received command for Endpoint=0 Cluster=0xFFF1_FC06 Command=0x0000_0000
[1660712608.337890][30773:30773] CHIP:ZCL: FaultInjection: Configure a fault of type: 1 and Id: 1 to be triggered deterministically

```
